wgpu-hal/src/dx12/command.rs

```diff
@@ -312,6 +312,15 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
                 barrier.usage.start,
                 barrier.usage.end
             );
+
+            if !barrier.buffer.initialized.swap(true, Ordering::SeqCst) {
+                unsafe {
+                    self.list
+                        .unwrap()
+                        .DiscardResource(barrier.buffer.resource.as_mut_ptr(), ptr::null());
+                }
```

Comment:
I think the only reason we're getting away with this is that discard resource is a no-op on buffers on most implementations. By way of the spec, this will clear all the data we just wrote into it.

Reply:
In my original commit, discard before writing data, would this be feasible?